osu.Game/Rulesets/UI/Scrolling/ScrollingHitObjectContainer.cs

```diff
@@ -215,16 +217,18 @@ protected override void Update()
 
                 foreach (var obj in Objects)
                 {
-                    if (!hitObjectInitialStateCache.TryGetValue(obj, out var objCache))
-                        objCache = hitObjectInitialStateCache[obj] = new Cached();
+                    if (!hitObjectInitialStateCache.TryGetValue(obj, out var state))
+                        state = hitObjectInitialStateCache[obj] = new InitialState(new Cached());
 
-                    if (objCache.IsValid)
+                    if (state.Cache.IsValid)
                         continue;
 
+                    state.ScheduledComputation?.Cancel();
+                    state.ScheduledComputation = computeInitialStateRecursive(obj);
```

Comment:
this is really hard to follow... might help to have some inline comments explaining what this is doing